From 090ac46912025f94aca710355e51a8154df36a56 Mon Sep 17 00:00:00 2001
From: Niccol Venerandi <niccolo@venerandi.com>
Date: Wed, 11 Mar 2020 18:46:07 +0100
Subject: [PATCH] Added support for libgit>=.99

Summary: LIBGIT2_SOVERSION no longer returns an integer in libgit>=.99; this adds support for that version and 1.0 to come.

Differential Revision: https://phabricator.kde.org/D27990
---
 src/application.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/application.cpp b/src/application.cpp
index c52fb574..b6738489 100644
--- a/src/application.cpp
+++ b/src/application.cpp
@@ -56,7 +56,7 @@ Application::Application(int &argc, char **argv)
 
 
     #ifdef WITH_LIBGIT2
-        #if LIBGIT2_SOVERSION >= 22
+        #if LIBGIT2_VER_MAJOR > 0 || (LIBGIT2_VER_MAJOR == 0 && LIBGIT_VER_MINOR >= 22)
             git_libgit2_init();
         #else
             git_threads_init();
@@ -67,7 +67,7 @@ Application::Application(int &argc, char **argv)
 Application::~Application()
 {
     #ifdef WITH_LIBGIT2
-        #if LIBGIT2_SOVERSION >= 22
+        #if LIBGIT2_VER_MAJOR > 0 || (LIBGIT2_VER_MAJOR == 0 && LIBGIT_VER_MINOR >= 22)
             git_libgit2_shutdown();
         #else
             git_threads_shutdown();
-- 
GitLab

